framework/src/utils/lib/s3-data-copy.ts

```diff
@@ -0,0 +1,121 @@
+
+  /**
+   * The Lambda function used to copy the data
+   */
+  public readonly copyLambda: IFunction;
+  /**
+   * The IAM role used by the lambda to copy the data
+   */
+  public readonly executionRole: IRole;
+
+
+  constructor(scope: Construct, id: string, props: S3DataCopyProps) {
+
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: S3DataCopy.name,
+    };
+    super(scope, id, trackedConstructProps);
+
+    if (props.vpc === undefined && props.subnets !==undefined) {
+      throw new Error('S3DataCopy error: if VPC parameter is not configured, subnets cannot be');
+    }
+    if (props.vpc !== undefined && props.subnets === undefined) {
+      throw new Error('S3DataCopy error: if VPC parameter is configured, subnets must be');
+    }
+
+    const functionName = `s3-data-copy-${Utils.generateScopeHash(this)}`;
+
+    const managedPolicy = new ManagedPolicy(this, 'Policy', {
+      document: new PolicyDocument({
+        statements: [
+          new PolicyStatement({
+            effect: Effect.ALLOW,
+            actions: [
+              'logs:CreateLogGroup',
+              'logs:CreateLogStream',
+              'logs:PutLogEvents',
+            ],
+            resources: [
+              `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/lambda/${functionName}`,
+              `arn:aws:logs:${Aws.REGION}:${Aws.ACCOUNT_ID}:log-group:/aws/lambda/${functionName}:log-stream:*`,
+            ],
+          }),
+          new PolicyStatement({
+            actions: [
+              'ec2:CreateNetworkInterface',
+              'ec2:DescribeNetworkInterfaces',
+              'ec2:DeleteNetworkInterface',
+              'ec2:AssignPrivateIpAddresses',
+              'ec2:UnassignPrivateIpAddresses',
+            ],
+            effect: Effect.ALLOW,
+            resources: ['*'],
+          }),
+        ],
+      }),
+    });
+
+    this.executionRole = props.executionRole || new Role(this, 'Role', {
+      assumedBy: new ServicePrincipal('lambda.amazonaws.com'),
+      description: 'Role used by S3DataCopy to copy data from one S3 bucket to another',
+    });
+    this.executionRole.addManagedPolicy(managedPolicy);
+
+    props.sourceBucket.grantRead(this.executionRole, `${props.sourceBucketPrefix || ''}*`);
+    props.targetBucket.grantWrite(this.executionRole, `${props.targetBucketPrefix || ''}*`);
+
+    this.copyLambda = new NodejsFunction(this, 'Lambda', {
+      runtime: S3DataCopy.CR_RUNTIME,
+      role: this.executionRole,
+      entry: path.join(__dirname, './resources/lambda/s3-data-copy/index.mjs'),
+      timeout: Duration.minutes(15),
+      environment: {
+        SOURCE_BUCKET_NAME: props.sourceBucket.bucketName,
+        SOURCE_BUCKET_PREFIX: props.sourceBucketPrefix || '',
+        SOURCE_BUCKET_REGION: props.sourceBucketRegion,
+        TARGET_BUCKET_NAME: props.targetBucket.bucketName,
+        TARGET_BUCKET_PREFIX: props.targetBucketPrefix || '',
```

Comment:
The character here is not a space but a "no-break space NBSP" (Unicode U+00A0). We shouldn't use special characters in code.
```suggestion
        TARGET_BUCKET_PREFIX: props.targetBucketPrefix || '',
```